pylint/checkers/variables.py

```diff
@@ -1326,9 +1327,23 @@ def visit_for(self, node: nodes.For) -> None:
             if any(isinstance(target, nodes.Starred) for target in targets):
                 return
 
-        if len(targets) != len(values):
-            details = _get_unpacking_extra_info(node, inferred)
-            self._report_unbalanced_unpacking(node, inferred, targets, values, details)
+        if isinstance(inferred, nodes.Dict) and isinstance(node.iter, nodes.Name):
+            # If this a case of 'dict-items-missing-iter', we don't want to
+            # report it as an 'unbalanced-dict-unpacking' as well
+            # TODO (performance), merging both checks would streamline this
+            if len(targets) == 2:
+                return
+
+        is_stared_targets = any(isinstance(target, nodes.Starred) for target in targets)
+        for value in values:
```

Comment:
Don't we want to test for cases like `for i,j,k in {(a,b):1, (d,e): 2}.keys()`?

Reply:
Don't we want to test for cases like `for i,j in {(a,b,c):1, (d,e,f): 2}.keys()` ?

Reply:
Yeah then we could also check for `nodes.DictKeys` perhaps?
I think it would be consistent with the other code above where we have a block of code specifically for `DictItems`: `if isinstance(inferred, astroid.objects.DictItems):` and now we would be explicitly dealing with these other situations `items()` and `keys()`.

Reply:
Got it, I'll refactor the code to be explicit about the cases each branch covers